controller.go

```diff
@@ -233,6 +257,121 @@ func (c *controller) makeDriverConfig(ntype string) map[string]interface{} {
 	return config
 }
 
+func (c *controller) registerNetwork(n Network) error {
+	if err := c.addNetwork(n.(*network)); err != nil && !strings.Contains(err.Error(), "exists") {
```

Comment:
This should not be needed. On daemon restart the networks are supposed to be present already.
Are you trying to populate some in-memory states with this ?